platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -366,4 +422,67 @@ private int findJoystickDevice(int device_id) {
 
 		return -1;
 	}
+
+	private boolean handleMouseEvent(final MotionEvent event) {
+		switch (event.getActionMasked()) {
+			case MotionEvent.ACTION_HOVER_ENTER:
+			case MotionEvent.ACTION_HOVER_MOVE:
+			case MotionEvent.ACTION_HOVER_EXIT: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int type = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.hover(type, x, y);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_BUTTON_PRESS:
+			case MotionEvent.ACTION_BUTTON_RELEASE:
+			case MotionEvent.ACTION_MOVE: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int buttonsMask = event.getButtonState();
+				final int action = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.touch(event.getSource(), action, 0, 1, new float[] { 0, x, y }, buttonsMask, 1);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_SCROLL: {
+				final float[] positionArray = new float[] { 0, event.getX(), event.getY() };
+
+				if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), event.getButtonState(), MOUSE_WHEEL_UP, event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) < 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), event.getButtonState(), MOUSE_WHEEL_DOWN, -event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				}
+
+				if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), event.getButtonState(), MOUSE_WHEEL_RIGHT, event.getAxisValue(MotionEvent.AXIS_HSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) < 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), event.getButtonState(), MOUSE_WHEEL_LEFT, -event.getAxisValue(MotionEvent.AXIS_HSCROLL), positionArray);
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	private void handleMouseScrollWheelButtonEvent(final int source, final int buttonsMask, final int scrollButton, final float factor, final float[] positionArray) {
+		queueEvent(new Runnable() {
+			@Override
+			public void run() {
+				int buttonsMaskWithScrollButton = buttonsMask | scrollButton;
```

Comment:
The value for `MOUSE_WHEEL_UP` currently collides with `BUTTON_STYLUS_SECONDARY`
https://developer.android.com/reference/android/view/MotionEvent#BUTTON_STYLUS_SECONDARY

I still think it would be safer to pass wheel buttons separately instead of mixing our own enum values with an Android enum so our code doesn't rely on values from the Android SDK.

Reply:
I know, that's exactly what I thought about (I don't really care about colliding with stylus primary as I check to see if the input is mouse, but I understand your reasoning, it's still wip as I wanted you comment about it and but first I wanted to see that it works)